ui/src/app/modules/auth/auth-interceptor/auth.interceptor.spec.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2021 Ford Motor Company
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { AuthInterceptor } from './auth-interceptor.service';
+import { AuthService } from '../auth.service';
+import { Router } from '@angular/router';
+import { of, throwError } from 'rxjs';
+import { HttpErrorResponse } from '@angular/common/http';
+import { isEmpty } from 'rxjs/operators';
+
+describe('AuthInterceptor', () => {
+  const fakeToken = 'fake-token';
+
+  let interceptor;
+  let mockHttpRequest;
+  let mockRouter: Router;
+  let mockHttpHandler;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    mockHttpRequest = { url: '', clone: jest.fn() };
+    mockRouter = { navigateByUrl: jest.fn() } as unknown as Router;
+    mockHttpHandler = { handle: jest.fn(() => of({})) };
+
+    const spiedGetToken = jest.spyOn(AuthService, 'getToken');
+    spiedGetToken.mockImplementation(() => fakeToken);
```

Comment:
You could just as easily write `AuthService.getToken = jest.fn().mockReturnValue(fakeToken)`, no?

Reply:
Will fix